log4j-api/src/main/java/org/apache/logging/log4j/message/ReusableSimpleMessage.java

```diff
@@ -43,7 +43,7 @@ public String getFormattedMessage() {
 
     @Override
     public String getFormat() {
-        return getFormattedMessage();
+        return charSequence instanceof String ? (String) charSequence : null;
```

Comment:
This is wrong. If charSequence is a StringBuilder or CharBuffer, the method would return null after the proposed change.